drivers/pinctrl/pinctrl_iopctl.c

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2024 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT nxp_iopctl
+
+#include <zephyr/drivers/pinctrl.h>
+#include <zephyr/init.h>
+
+#define OFFSET(mux) (((mux) & 0xFFF00000) >> 20)
+#define INDEX(mux)  (((mux) & 0xF0000) >> 16)
+
+/* Port register addresses. */
+static uint32_t *iopctl[] = {
+	(uint32_t *)DT_REG_ADDR(DT_NODELABEL(iopctl0)),
+#if DT_NUM_INST_STATUS_OKAY(DT_DRV_COMPAT) > 1
+	(uint32_t *)DT_REG_ADDR(DT_NODELABEL(iopctl1)),
+#endif
+#if DT_NUM_INST_STATUS_OKAY(DT_DRV_COMPAT) > 2
```

Comment:
why not just check if node exists? this looks fragile

Reply:
Updated. Thank you.